src/config.c

```diff
@@ -45,6 +45,12 @@ typedef struct configEnum {
     const int val;
 } configEnum;
 
+typedef struct deprecatedConfig {
+    const char* name;
+    const int argc_min;
+    const int argc_max;
+} deprecatedConfig;
```

Comment:
This is good, but I wonder if we really need to store the argc info for deprecated configs. It might be enough just to know the deprecated config's name in order to ignore it since we don't really case what args are passed to it.

Reply:
i feel we're devoting too much code to this "mechanism", personally i'm also ok with the previous approach (as long as we don't have too many dead configs, e.g. more than 8)